### PR TITLE
Add validation test for groovy.transform.Immutable

### PR DESCRIPTION
**Pull request by ollin**

### CODE REVIEW

File: org.jacoco.core.test.validation.groovy/src/org/jacoco/core/test/validation/groovy/GroovyImutableClassTest.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *    Oliver Nautsch - initial API and implementation
+ *
+ *******************************************************************************/
+package org.jacoco.core.test.validation.groovy;
+
+import org.jacoco.core.test.validation.ValidationTestBase;
+import org.jacoco.core.test.validation.groovy.targets.GroovyDataClassTarget;
```
Comment (Godin) [inline org.jacoco.core.test.validation.groovy/src/org/jacoco/core/test/validation/groovy/GroovyImutableClassTest.java]: Unused import.

File: org.jacoco.core.test.validation.groovy/src/org/jacoco/core/test/validation/groovy/GroovyImutableClassTest.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+import org.jacoco.core.test.validation.groovy.targets.GroovyImutableClassTarget;
+import org.junit.Test;
+
+/**
+ * Test of <code>data class</code>es.
```
Comment (Godin) [inline org.jacoco.core.test.validation.groovy/src/org/jacoco/core/test/validation/groovy/GroovyImutableClassTest.java]: Seems that change of comment was forgotten after copy-paste from `GroovyDataClassTest`.

**Comment by Godin** [comment]
@ollin thank you for one more contribution! and @aalmiray thank you again for Hackergarten! ❤️
I'll make necessary cleanup before merge.

File: org.jacoco.core.test.validation.groovy/pom.xml [hunk 3]
```diff
@@ -26,7 +26,7 @@
 
   <properties>
     <gmaven.version>1.6.2</gmaven.version>
-    <groovy.version>2.5.3</groovy.version>
+    <groovy.version>2.5.6</groovy.version>
```
Comment (aalmiray) [inline org.jacoco.core.test.validation.groovy/pom.xml]: FYI Groovy 2.5.7 was just released, fixing some issues reported during the Hackergarten

**Comment by aalmiray** [comment]
Thanks! 😄

File: org.jacoco.core.test.validation.groovy/pom.xml (hunk 3 above)
Comment (Godin) [reply org.jacoco.core.test.validation.groovy/pom.xml]: I know, and new version also works with JDK 13 - https://github.com/jacoco/jacoco/commit/1a476d57546c3f74fb4daa31c04464eafde935d8 ;)

**Pull request merged**